### ISSUE TITLE
Login page changes breaks proxy configuration

### ISSUE THREAD

**Issue by KeithB** [pi-hole/web#2488]
### Context
This ticket describes how a Pi-Hole change breaks previously working end-to-end functionality in a non-vanilla environment together with a potential solution to accommodate - as a vanilla standalone install there is no bug. 

Short version - would the patch below be considered for inclusion? (or a variation if there are coding styles etc missing) Happy to adjust to a PR from a fork if needed.

### Versions
  Pi-hole version is v5.14.2 (Latest: v5.14.2)
  AdminLTE version is v5.18 (Latest: v5.18)
  FTL version is v5.20 (Latest: v5.20)

### Platform
Running in official Pi-Hole container on docker on Rpi 64 bit Bullseye.

Linux piccpu3 5.15.76-v8+ #1597 SMP PREEMPT Fri Nov 4 12:16:41 GMT 2022 aarch64 GNU/Linux
Docker version 20.10.22, build 3a2c30b

### Additional configuration info
- Running 1 Pi-Hole container per Pi across multiple Pis.
- Nginx proxy running on separate container fronting to the UI. Each UI configured for forwarding from `dns.mydomain.com/pihole#` to `rPi#:8081/admin`.
- Base URL of `dns.mydomain.com` routes to `rPi1:8081/admin`. This allows simple access to configure a single Pi-Hole instance and leave other sync functionality to mirror it in to the other instances.

### Expected behavior
Before the login page changes, the proxy configuration with Pi-Hole took the user to login and after successful login it re-directed to `rPi#:8081/admin/index.php`. This gave correct access through the various Pi-Hole instances.

### Actual behavior / bug
After the login page changes,  successful login re-directs to `mydomain.com/admin/`. So login to Pi-Hole #3 redirects to /admin/ which the browser then sees as a re-direct to `dns.mydomain.com/admin` and because of the nginx config gets seen by the Pi-Hole instances as `rPi1:8081/admin/admin`. 

This fails but going back to the original `dns.mydomain.com/pihole#` URL then succeeds as the hard-coded index.php redirect kicks in (see next paragraph).

Running through the code changes, the original login page was fixed to redirect to index.php (and it still does if an authenticated session is in place) but the new changes pick up the original URL and attempt a re-direct. When there is a proxy or similar between the browser and Pi-Hole that manipulates the URL this fails - the browser has no idea of the Pi-Hole setup and Pi-Hole has no idea of the original URL called.

More discussion/description at https://discourse.pi-hole.net/t/did-the-login-url-handling-change/60385

### Potential solution
Some applications given configuration options to support proxy setup - base URLs etc.

Given all URLs for the portal are pre-fixed by /admin, the following patch is an option to allow an optional config option (WEBADMINROOT) to be added to setupVars.conf. If included the re-direct is amended to consider this becoming more proxy friendly.

```
--- password.php.backup	2023-01-10 22:22:16.498562202 +0000
+++ password.php	2023-01-11 22:26:22.834883397 +0000
@@ -21,7 +21,14 @@
     $pwhash = '';
 }
 
-function verifyPassword($pwhash, $use_api = false)
+// Try to read admin root from setupVars.conf
+if (isset($setupVars['WEBADMINROOT'])) {
+    $webadminroot = $setupVars['WEBADMINROOT'];
+} else {
+    $webadminroot = '';
+}
+
+function verifyPassword($pwhash, $use_api = false, $webadminroot)
 {
     $validpassword = true;
 
@@ -48,6 +55,11 @@
                 $redirect_url = 'index.php';
                 if (isset($_SESSION['prev_url'])) {
                     $redirect_url = $_SESSION['prev_url'];
+                    if ( strlen($webadminroot) > 0) {
+                        if (substr($redirect_url, 0, 6) === '/admin') {
+                            $redirect_url=$webadminroot.substr($redirect_url,6);
+                        }
+                    }
                 }
 
                 // Regenerate session ID to prevent session fixation
@@ -98,5 +110,5 @@
     return $validpassword;
 }
 
-$wrongpassword = !verifyPassword($pwhash, isset($api));
+$wrongpassword = !verifyPassword($pwhash, isset($api), $webadminroot);
 $auth = $_SESSION['auth'];

```

**Comment by luiznaac** [comment]
@KeithB did you manage to get this working in any other way? I'm facing the exactly same issue here

**Comment by KeithB** [comment]
> @KeithB did you manage to get this working in any other way? I'm facing the exactly same issue here

Hi @luiznaac . 

I spent a bit of time looking at it and potential code changes but it introduced other challenges. In the end it wasn't worth the effort as there were other changes needed and I thought it more likely to cause more harm than good. 

I swapped to per instance hostnames - currently running dns1.x.y.z and dns2.x.y.z as hostnames for each instance.
